src/gui/processing/qgsprocessingwidgetwrapper.cpp

```diff
@@ -366,7 +366,7 @@ void QgsAbstractProcessingParameterWidgetWrapper::setDynamicParentLayerParameter
 QgsProcessingModelerParameterWidget *QgsProcessingParameterWidgetFactoryInterface::createModelerWidgetWrapper( QgsProcessingModelAlgorithm *model, const QString &childId, const QgsProcessingParameterDefinition *parameter, QgsProcessingContext &context )
 {
   auto widget = std::make_unique<QgsProcessingModelerParameterWidget>( model, childId, parameter, context );
-  widget->populateSources( compatibleParameterTypes(), compatibleOutputTypes(), compatibleDataTypes( parameter ) );
+  widget->populateSources( parameter );
```

Comment:
As noted elsewhere, we'll need to keep old behavior here for api compatibility (as QgsAbstractProcessingParameterWidgetWrapper IS part of stable API)